models/DataObject/Data/UrlSlug.php

```diff
@@ -132,11 +137,33 @@ public function getSlug(): ?string
      */
     public function setSlug(?string $slug)
     {
+        if($this->slug && !$this->previousSlug && $slug !== $this->slug) {
+            $this->previousSlug = $this->slug;
```

Comment:
wouldn't it be better to do this in createFromDataRow?

Reply:
But there it's not covering API updates ... so I guess this is the best place, right?

Reply:
Would still say that previousSlug should only be set when loaded from resource. It should be kind of a dirty detection, which causes a redirect, right ?

Reply:
Yep, now I got your point 😉